components/backup/src/endpoint.rs

```diff
@@ -1512,6 +1562,197 @@ pub mod tests {
         }
     }
 
+    #[test]
+    fn test_seek_ranges() {
+        let (_tmp, endpoint) = new_endpoint();
+
+        endpoint.region_info.set_regions(vec![
+            (b"".to_vec(), b"1".to_vec(), 1),
+            (b"1".to_vec(), b"2".to_vec(), 2),
+            (b"3".to_vec(), b"4".to_vec(), 3),
+            (b"7".to_vec(), b"9".to_vec(), 4),
+            (b"9".to_vec(), b"".to_vec(), 5),
+        ]);
+        // Test seek backup range.
+        let test_seek_backup_ranges =
+            |sub_ranges: Vec<(&[u8], &[u8])>, expect: Vec<(&[u8], &[u8])>| {
+                let mut ranges = Vec::with_capacity(sub_ranges.len());
+                for &(start_key, end_key) in &sub_ranges {
+                    let start_key = if start_key.is_empty() {
+                        None
+                    } else {
+                        Some(Key::from_raw(start_key))
+                    };
+                    let end_key = if end_key.is_empty() {
+                        None
+                    } else {
+                        Some(Key::from_raw(end_key))
+                    };
```

Comment:
We can replace it with [`bool::then`](http://doc.rust-lang.org/1.65.0/core/primitive.bool.html#method.then)